airflow/jobs/scheduler_job_runner.py

```diff
@@ -1367,7 +1367,7 @@ def _create_dag_runs(self, dag_models: Collection[DagModel], session: Session) -
                         data_interval=data_interval,
                         external_trigger=False,
                         session=session,
-                        dag_hash=dag_hash,
+                        dag_version_id=latest_dag_version.id if latest_dag_version else None,
```

Comment:
similar here, the if / else is clunky

Reply:
I have removed this and adjusted the tests. I didn't want to make many edits on the scheduler tests, that was why I had this but I have corrected it